src/Microsoft.AspNet.Razor/CodeGenerators/CSharpTagHelperCodeRenderer.cs

```diff
@@ -179,68 +177,45 @@ protected virtual string GenerateUniqueId()
                                                           _tagHelperContext.ExecutionContextAddMethodName,
                                                           tagHelperVariableName);
                 }
-
-                // Render all of the bound attribute values for the tag helper.
-                RenderBoundHTMLAttributes(
-                    chunk.Attributes,
-                    tagHelperDescriptor.TypeName,
-                    tagHelperVariableName,
-                    tagHelperDescriptor.Attributes,
-                    htmlAttributeValues);
             }
         }
 
-        private void RenderBoundHTMLAttributes(
-            IList<KeyValuePair<string, Chunk>> chunkAttributes,
-            string tagHelperTypeName,
-            string tagHelperVariableName,
-            IEnumerable<TagHelperAttributeDescriptor> attributeDescriptors,
-            Dictionary<string, string> htmlAttributeValues)
+        private void RenderTagHelperPropertyVerification(
+            IEnumerable<KeyValuePair<string, Chunk>> chunkAttributes,
+            IEnumerable<TagHelperDescriptor> tagHelperDescriptors)
         {
+            if (_designTimeMode)
+            {
+                // No need to render any dictionary verification in design time mode.
+                return;
+            }
+
             // Track dictionary properties we have confirmed are non-null.
             var confirmedDictionaries = new HashSet<string>(StringComparer.Ordinal);
```

Comment:
Added this change then regenerated the code and your dictionary tests actually do cover this scenario but the old output was wrong. In the commit to address code review comments you can see the change for the added throw cases.